src/prefect/agent/agent.py

```diff
@@ -118,27 +118,45 @@ def start(self) -> None:
         The main entrypoint to the agent. This function loops and constantly polls for
         new flow runs to deploy
         """
-        with exit_handler(self) as exit_event:
-            tenant_id = self.agent_connect()
-
-            # Loop intervals for query sleep backoff
-            loop_intervals = {0: 0.25, 1: 0.5, 2: 1.0, 3: 2.0, 4: 4.0, 5: 8.0, 6: 10.0}
+        try:
+            with exit_handler(self) as exit_event:
+                tenant_id = self.agent_connect()
+
+                # Loop intervals for query sleep backoff
+                loop_intervals = {
+                    0: 0.25,
+                    1: 0.5,
+                    2: 1.0,
+                    3: 2.0,
+                    4: 4.0,
+                    5: 8.0,
+                    6: 10.0,
+                }
 
-            index = 0
-            while not exit_event.wait(timeout=loop_intervals[index]):
-                self.heartbeat()
+                index = 0
+                while not exit_event.wait(timeout=loop_intervals[index]):
+                    self.heartbeat()
 
-                runs = self.agent_process(tenant_id)
-                if runs:
-                    index = 0
-                elif index < max(loop_intervals.keys()):
-                    index += 1
+                    runs = self.agent_process(tenant_id)
+                    if runs:
+                        index = 0
+                    elif index < max(loop_intervals.keys()):
+                        index += 1
 
-                self.logger.debug(
-                    "Next query for flow runs in {} seconds".format(
-                        loop_intervals[index]
+                    self.logger.debug(
+                        "Next query for flow runs in {} seconds".format(
+                            loop_intervals[index]
+                        )
                     )
-                )
+        finally:
+            self.on_shutdown()
+
+    def on_shutdown(self) -> None:
```

Comment:
Question: do you think this shutdown hook is worth of a changelog update?

Reply:
Discussed offline, no need for an update

Reply:
completely subjective: I vote no. The hook isn't being used by any other agent right now thus doesn't modify any existing behavior. If this was a notable hook in another way I might be inclined to say otherwise.